halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/model/v1/security/Authn.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2017 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.netflix.spinnaker.halyard.config.model.v1.security;
+
+import com.netflix.spinnaker.halyard.config.model.v1.node.Node;
+import com.netflix.spinnaker.halyard.config.model.v1.node.NodeIterator;
+import com.netflix.spinnaker.halyard.config.model.v1.node.NodeIteratorFactory;
+import com.netflix.spinnaker.halyard.config.model.v1.node.Validator;
+import com.netflix.spinnaker.halyard.config.problem.v1.ConfigProblemSetBuilder;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+@Data
+@EqualsAndHashCode(callSuper = false)
+public class Authn extends Node {
+  @Override
+  public void accept(ConfigProblemSetBuilder psBuilder, Validator v) {
```

Comment:
`accept`: AFAIK the visitor pattern only works like this (to make sure the right class type is passed to the validator).

`getChildren` The thing is, the common implementation (empty iterator) is very wrong in many cases. I want to draw attention to the fact that this needs to be implemented for any new contributors so they don't get bitten by an incorrect (although typically functional) default.